lamindb/_artifact.py

```diff
@@ -874,6 +874,7 @@ def replace(
         if key_path.name != new_filename:
             self._clear_storagekey = self.key
             self.key = str(key_path.with_name(new_filename))
+            self._key = self.key
```

Comment:
@sunnyosun @falexwolf this is so so confusing, why calling it _key and not _old_key or _previous_key?

Reply:
You're absolutely right, @Koncopd! I should have pointed this out to. We over-use "simply prefixing with _".

Reply:
Can this be fixed, please?

Reply:
I see it was fixed already somewhere.